Bindings/Java/Matlab/Utilities/osimSimulate.m

```diff
@@ -1,12 +1,30 @@
-function Simulate(model, state, visualize)
-% Simulate an OpenSim model from an initial state. The provided state is
-% updated to be the state at the end of the simulation.
+function osimSimulate(model, stateOrVisualize, finalTime)
+% Simulate an OpenSim model.
+%
+% The simulation can be run with or without visualization (see stateOrVisualize
+% for more information). If visualizing (with the simbody-visualizer), you will
+% be able run the simulation multiple times.
+% Otherwise, the simulation is run once without visualization.
+%
+% Use Ctrl-C in the command window to halt the simulation.
+%
 %
 % Parameters
 % ----------
-% model: The OpenSim Model to simulate.
-% state: The SimTK State to use as the initial state for the simulation.
-% visualize (bool): Use the simbody-visualizer to visualize the simulation?
+% model: The OpenSim Model to simulate. You must have called initSystem() on
+%        this model.
+% stateOrVisualize:
+%    State mode: The second parameter is the SimTK State to use as the initial
+%        state for the simulation. In this case, you must have already called
+%        initSystem() on the model. The provided state is updated to be the
+%        state at the end of the simulation.
+%    Visualize mode: The second parameter is a boolean indicating whether or
+%        not the visualizer should be used. In this case, you need not call
+%        initSystem() yourself. This function will call initSystem()
+%        internally, and the simulation will start from the default state. In
+%        this mode, this parameter is overridden by the OPENSIM_USE_VISUALIZER
+%        environment variable, if set.
```

Comment:
I know this parameter is odd, but it allowed me to do the following:

1. Allow the use of an environment variable, `OPENSIM_USE_VISUALIZER`, to override the visualizer, so that some Matlab scripts can undergo automated testing.
2. Allow users to provide a custom initial state.

However, these two are not possible simultaneously (hence the `stateOrVisualize` parameter). 

- Allowing an override to the visualizer means that we must call `initSystem()` *within* `osimSimulate()`.
- Allowing users to provide a custom initial state means we *cannot* call `initSystem()` within `osimSimulate()`. The specific issue I ran into is related to locking a coordinate. Given a state, a user can lock a coordinate at any value, but once `initSystem()` is called, the coordinate is locked at the *default* value (the locked-at value is stored in the state, not the model, and there is no way to recover the locked-at value during `initSystem()` unless we create a property for it). Therefore, calling `initSystem()` causes us to not fully obey the initial state the user provided.